core/src/main/java/com/sequenceiq/distrox/v1/distrox/StackOperations.java

```diff
@@ -258,24 +262,24 @@ public FlowIdentifier updatePillarConfiguration(@NotNull NameOrCrn nameOrCrn, Lo
         return clusterCommonService.updatePillarConfiguration(nameOrCrn, workspaceId);
     }
 
-    public UpgradeV4Response checkForClusterUpgrade(@NotNull Stack stack, Long workspaceId, UpgradeV4Request request) {
+    public UpgradeV4Response checkForClusterUpgrade(String accountId, @NotNull Stack stack, Long workspaceId, UpgradeV4Request request) {
         MDCBuilder.buildMdcContext(stack);
         boolean osUpgrade = upgradeService.isOsUpgrade(request);
         boolean replacevms = determineReplaceVmsParameter(stack, request.getReplaceVms());
         UpgradeV4Response upgradeResponse = clusterUpgradeAvailabilityService.checkForUpgradesByName(stack, osUpgrade, replacevms);
         if (CollectionUtils.isNotEmpty(upgradeResponse.getUpgradeCandidates())) {
             clusterUpgradeAvailabilityService.filterUpgradeOptions(upgradeResponse, request, stack.isDatalake());
         }
-        validateDatalakeHasNoRunningDatahub(workspaceId, stack, upgradeResponse);
+        validateDatalakeHasNoRunningDatahub(accountId, workspaceId, stack, upgradeResponse);
         return upgradeResponse;
     }
 
-    public UpgradeV4Response checkForClusterUpgrade(@NotNull NameOrCrn nameOrCrn, Long workspaceId, UpgradeV4Request request) {
-        return checkForClusterUpgrade(stackService.getByNameOrCrnInWorkspace(nameOrCrn, workspaceId), workspaceId, request);
+    public UpgradeV4Response checkForClusterUpgrade(String accountId, @NotNull NameOrCrn nameOrCrn, Long workspaceId, UpgradeV4Request request) {
+        return checkForClusterUpgrade(accountId, stackService.getByNameOrCrnInWorkspace(nameOrCrn, workspaceId), workspaceId, request);
     }
 
-    private void validateDatalakeHasNoRunningDatahub(Long workspaceId, Stack stack, UpgradeV4Response upgradeResponse) {
-        if (StackType.DATALAKE == stack.getType()) {
+    private void validateDatalakeHasNoRunningDatahub(String accountId, Long workspaceId, Stack stack, UpgradeV4Response upgradeResponse) {
+        if (entitlementService.runtimeUpgradeEnabled(accountId) && StackType.DATALAKE == stack.getType()) {
```

Comment:
Can you please add a log message that we're checking for the Data Hubs, because you can do any kinds of upgrades?